### ISSUE TITLE
dotnet 2 solutions with more than 1 project?

### ISSUE THREAD

**Issue by dougtoppin** [redhat-developer/s2i-dotnetcore#149]
I am not a real dotnet person but doing a little with it with s2i so I may be missing something regarding building dotnet 2.0 solutions (using the Visual Studio vernacular) that have more than one project in them.
From looking at assemble there is a find looking for proj files but set to a maxdepth of 1 at https://github.com/redhat-developer/s2i-dotnetcore/blob/4759b50add74a2871afcefd5e94d2db0abdb80a1/2.0/build/s2i/bin/assemble#L46

I think it then works like this:

* 1 project found, continues as normal
* 0 projects found, error
* more than 1 project found, it suggests selecting 1 in a .s2i/environment file to specify one of them

The tests all appear to be using single projects.

My question is should assemble support solutions that contain multiple projects (which I think is the normal dotnet architecture)?
If so, are there examples of how to do this?

Again, I may be completely off base and missing something entirely but the s2i related documentation does not appear to address this.

**Comment by omajid** [comment]
I think we need to step back and see how this is meant to be used. The idea is that s2i-dotnetcore (and s2i in general) will build and run some code. It can only run one thing at a time. Running two things simultaneously isn't an option (you need to use two different s2i invocations and/or two different containers for that). So we just want to build enough to run that one piece of code. As  you say, it is common to have multiple projects in a .NET Core solution. So we really just want to build the target project that we are interested in running, as well as any dependencies it has. The dependencies should be built automatically if they are a project to project dependency (that is, the project we want to build and run says it depends on another project's source in the solution) or pulled down from nuget.org and used.

Can you think of a situation where we want to build two different/unrelated projects and only run one?

**Comment by dougtoppin** [comment]
I see my confusion now. It is a top-level .sln with 4 projects under it. 3 of the projects produce dll files and are each runnable by dotnet. The 4th is a shared library used by one of the projects. I need to have 3 pipelines which build and deploy each of the 3 projects independently.
I just need to figure out how to set the env vars to the correct directories for the 3 builds.
When I checked the tests I only saw one project in each and that is where I got stuck.

Sorry about the trouble.
